src/main/java/domain/board/Board.java

```diff
@@ -0,0 +1,153 @@
+package domain.board;
+
+import domain.piece.Piece;
+import domain.piece.PieceType;
+import domain.piece.Team;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Board {
+
+    private final Map<Point, Piece> pieceByPoint;
+    private final PointNodeMapper pointNodeMapper;
+
+    public Board(final Map<Point, Piece> pieceByPoint, final PointNodeMapper pointNodeMapper) {
+        this.pieceByPoint = pieceByPoint;
+        this.pointNodeMapper = pointNodeMapper;
+    }
+
+    public boolean isEnd() {
+        return !isTwoWangsAlive();
+    }
+
+    private boolean isTwoWangsAlive() {
+        return findTeamsOfWang().containsAll(List.of(Team.CHO, Team.HAN));
```

Comment:
추가적으로 워딩을 생각했을 때, 아래와 같은 로직이 네이밍이랑 조금 더 매칭이 잘되지 않을까요? 🤔 

```suggestion
        findTeamsOfWang().size() == 2;
```

Reply:
장기의 원래 룰대로라면 각 팀의 왕은 1개씩만 존재해야 하지만, 아래와 같은 예외 상황이 발생할 수 있다고 생각해서 팀의 종류까지 확인하게 했어요! 
* 한 팀의 왕이 2개 이상이 존재할 수 있는 경우